x-pack/solutions/security/plugins/security_solution/server/endpoint/services/actions/clients/microsoft/defender/endpoint/ms_defender_endpoint_actions_client.ts

```diff
@@ -1309,4 +1317,70 @@ export class MicrosoftDefenderEndpointActionsClient extends ResponseActionsClien
 
     return agentResponse;
   }
+
+  private async fetchMachineLiveResponseFile(
+    machineActionId: string
+  ): Promise<ResponseActionRunScriptOutputContent | null> {
+    try {
+      // Download the file content
+      const { data: downloadStream } = await this.sendAction<Readable>(
+        MICROSOFT_DEFENDER_ENDPOINT_SUB_ACTION.GET_ACTION_RESULTS,
+        { id: machineActionId }
+      );
+
+      if (!downloadStream) {
+        this.log.debug(`No download stream available for machine action ${machineActionId}`);
+        return null;
+      }
+
+      if (downloadStream.readableLength > RUNSCRIPT_OUTPUT_FILE_MAX_SIZE_BYTES) {
+        throw new ResponseActionsClientError(
+          'The output file is too large to download and display.',
+          413
+        );
+      }
+
+      return this.getFormattedOutput(downloadStream);
+    } catch ({ message, statusCode }) {
+      this.log.error(
+        `Failed to fetch runscript output for machine action ${machineActionId}: ${message}`
+      );
+
+      // Return error information as output
+      return {
+        stdout: '',
+        stderr: message,
+        code: statusCode,
+      };
+    }
+  }
+
+  private async getFormattedOutput(
+    stream: Readable
+  ): Promise<ResponseActionRunScriptOutputContent> {
+    try {
+      const buffer: Buffer[] = [];
+      const stderr = '';
+
+      for await (const chunk of stream) {
+        buffer.push(Buffer.isBuffer(chunk) ? chunk : Buffer.from(chunk));
+      }
+      const output = JSON.parse(Buffer.concat(buffer).toString('utf8'));
+
+      return {
+        stdout: output.script_output ?? '',
+        stderr: (output.script_errors ?? '') + (stderr ? `\n${stderr}` : ''),
+        code: output.exit_code ?? 0,
+      };
+    } catch ({ message }) {
+      const error = `Failed to process runscript output file: ${message}`;
+      this.log.error(error);
```

Comment:
can we improve the logging of the error here. Here is what I suggest:

change `catch ({ message }) {` to `catch (err)`

Log: `this.log.error(error, { error: err }`

This will also log the root error which will be useful in trying to figure  out where things went wrong